core/src/test/java/io/grpc/internal/SpiffeUtilTest.java

```diff
@@ -159,7 +159,7 @@ public void spiffeTrustDomainFormatTest() {
           SpiffeUtil.parse(longTrustDomain.toString()));
       assertEquals("Trust Domain maximum length is 255 characters", iae.getMessage());
 
-      StringBuilder longSpiffe = new StringBuilder("spiffe://mydomain%21com/");
+      StringBuilder longSpiffe = new StringBuilder(String.format("spiffe://mydomain%scom/", "%21"));
```

Comment:
How is this a style fix?  Did ErrorProne complain about this? I've not seen it complain about percent-encoded strings, and this change seems obviously worse than how it began.

Reply:
Yeah, it was a `String literal contains format specifiers, but is not passed to a format method` error. Agreed that a workaround looks really weird.
Is there any way to suppress such checks for a particular LOC?

Reply:
You can suppress warnings for a line that contains a declaration. `@SuppressWarnings("WarningName")`. If you follow the link that tells you more it tells you how to suppress. Looks like this one was https://errorprone.info/bugpattern/OrphanedFormatString . We should probably file a bug saying it was wrong, as percent encoding is pretty common, and that is clearly a URL. Looks like https://github.com/google/error-prone/issues/1021 is related, and the same logic could be applied to the StringBuilder constructor. I guess we normally don't hit this because we normally aren't using StringBuilder with URL and when we do, we're going to do something like `.append('%').append(toHex(c))` which wouldn't trigger it.